homeassistant/components/weatherflow_cloud/config_flow.py

```diff
@@ -0,0 +1,108 @@
+"""Config flow for WeatherflowCloud integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyweatherflow_forecast import (
+    WeatherFlow,
+    WeatherFlowForecastBadRequest,
+    WeatherFlowForecastInternalServerError,
+    WeatherFlowForecastUnauthorized,
+    WeatherFlowForecastWongStationId,
+    WeatherFlowStationData,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN, CONF_DEVICE_ID, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import (
+    _LOGGER,
+    CONF_FIRMWARE_REVISION,
+    CONF_SERIAL_NUMBER,
+    CONF_STATION_ID,
+    DOMAIN,
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlowCloud."""
+
+    VERSION = 1
+
+    async def _async_validate_and_get_station_info(
+        self, station_id: int, api_token: str
+    ) -> WeatherFlowStationData:
+        """Validate credentials and obtain station information."""
+
+        session = async_create_clientsession(self.hass)
+        api = WeatherFlow(station_id=station_id, api_token=api_token, session=session)
```

Comment:
```suggestion
        api = WeatherFlow(station_id, api_token, session)
```
Not sure if they are named arguments, but this is way better imo